src/apps/main/antivirus/ManualSystemScan.ts

```diff
@@ -0,0 +1,229 @@
+    this.scanSessionId++;
+    const currentSession = this.scanSessionId;
+
+    if (!this.antivirus) {
+      this.antivirus = await Antivirus.createInstance();
+    }
+    const antivirus = this.antivirus;
+
+    let reportProgressInterval: NodeJS.Timeout | null = null;
+
+    reportProgressInterval = setInterval(() => {
+      if (this.progressEvents.length > 0) {
+        eventBus.emit('ANTIVIRUS_SCAN_PROGRESS', {
+          ...(this.progressEvents.pop() as ProgressData),
+        });
+        this.progressEvents = [];
+      }
+    }, 1000);
+
+    console.time('manual-scan');
+
+    const scan = async (filePath: string) => {
+      if (this.cancelled) return;
+      console.log('SCAN ITEM: ', filePath);
+      try {
+        const scannedItem = await transformItem(filePath);
+        const previousScannedItem = await this.dbConnection.getItemFromDatabase(scannedItem.pathName);
+        if (previousScannedItem) {
+          await this.handlePreviousScannedItem(currentSession, scannedItem, previousScannedItem);
+          return;
+        }
+
+        const currentScannedFile = await antivirus.scanFile(scannedItem.pathName);
+
+        if (currentScannedFile) {
+          await this.dbConnection.addItemToDatabase({
+            ...scannedItem,
+            isInfected: currentScannedFile.isInfected,
+          });
+
+          this.trackProgress(currentSession, {
+            file: currentScannedFile.file,
+            isInfected: currentScannedFile.isInfected,
+          });
+        }
+      } catch (error) {
+        if (!isPermissionError(error)) {
+          throw error;
+        }
+      }
+    };
+
+    try {
+      if (!pathNames) {
+        const userSystemPath = await getUserSystemPath();
+        if (!userSystemPath) return;
+
+        this.totalItemsToScan = await countFilesUsingWindowsCommand(userSystemPath.path);
+
+        this.manualQueue = queue(scan, 10);
+
+        await getFilesFromDirectory(userSystemPath.path, (filePath: string) => this.manualQueue!.pushAsync(filePath));
+      } else {
+        let pathsToScan: string[] = [];
+        if (pathNames && pathNames.length > 0) {
+          pathsToScan = pathNames;
+        } else {
+          const userSystemPath = await getUserSystemPath();
+          if (!userSystemPath) return;
+          pathsToScan = [userSystemPath.path];
+        }
+        this.manualQueue = queue(scan, 10);
+
+        let total = 0;
+        for (const p of pathNames) {
+          total += await countFilesUsingWindowsCommand(p);
+        }
+
+        this.totalItemsToScan = total;
+
+        for (const p of pathsToScan) {
+          await getFilesFromDirectory(p, (filePath: string) => this.manualQueue!.pushAsync(filePath));
+        }
+      }
+
+      await this.manualQueue.drain();
+
+      this.finishScan(currentSession);
+    } catch (error) {
+      if (!isPermissionError(error)) {
+        throw error;
+      }
+    } finally {
+      console.timeEnd('manual-scan');
```

Comment:
The `console.time` for the entire process could be useful, but this one is going to overflow the entire logging system.

Reply:
This is used to obtain the time of the entire scan.